internal/storage/v2/clickhouse/tracestore/reader.go

```diff
@@ -129,9 +131,54 @@ func (*Reader) FindTraces(
 	panic("not implemented")
 }
 
-func (*Reader) FindTraceIDs(
-	context.Context,
-	tracestore.TraceQueryParams,
+func (r *Reader) FindTraceIDs(
+	ctx context.Context,
+	query tracestore.TraceQueryParams,
 ) iter.Seq2[[]tracestore.FoundTraceID, error] {
-	panic("not implemented")
+	return func(yield func([]tracestore.FoundTraceID, error) bool) {
+		q := sql.SearchTraceIDs
+		args := []any{}
+
+		if query.ServiceName != "" {
+			q += " AND service_name = ?"
+			args = append(args, query.ServiceName)
+		}
+		if query.OperationName != "" {
+			q += " AND name = ?"
+			args = append(args, query.OperationName)
+		}
+		if query.SearchDepth > 0 {
+			q += " LIMIT ?"
+			args = append(args, query.SearchDepth)
+		}
+
+		rows, err := r.conn.Query(ctx, q, args...)
+		if err != nil {
+			yield(nil, fmt.Errorf("failed to query trace IDs: %w", err))
+			return
+		}
+		defer rows.Close()
+
+		for rows.Next() {
+			var traceID string
+			if err := rows.Scan(&traceID); err != nil {
+				if !yield(nil, fmt.Errorf("failed to scan row: %w", err)) {
+					return
+				}
+				continue
+			}
+
+			b, err := hex.DecodeString(traceID)
+			if err != nil {
+				if !yield(nil, fmt.Errorf("failed to decode trace ID: %w", err)) {
+					return
+				}
+				continue
+			}
+
+			if !yield([]tracestore.FoundTraceID{{TraceID: pcommon.TraceID(b)}}, nil) {
```

Comment:
we should be returning timestamps with found trace IDs, this was one critical requirement why we added them to the GetTraces API because in CH finding trace by ID is not cheap without time range.